platform_plugin_forum_email_notifier/tasks.py

```diff
@@ -85,25 +134,150 @@ def notify_users(thread_id, discussion, course_id, body, title, url, object_type
+                ):
+                    continue
+            except User.DoesNotExist:
+                log.warning(f"User {subscriber} does not exist")
                 continue
-        except User.DoesNotExist:
-            log.warning(f"User {subscriber} does not exist")
-            return
+            except ForumNotificationPreference.DoesNotExist:
+                pass
+                # If the user does not have a preference and they already are a subscriber
+                # we should notify them
         send_email_notification.delay(
-            subscriber,
-            str(course_id),
+            thread_id,
+            discussion,
+            course_id,
             body,
             title,
             url,
+            author_id,
+            author_username,
+            author_email,
             object_type,
+            subscriber,
+        )
+
+    handle_digests.delay(
+        thread_id,
+        discussion,
+        course_id,
+        body,
+        title,
+        url,
+        author_id,
+        author_username,
+        author_email,
+        object_type,
+    )
+
+
+@shared_task
+@set_code_owner_attribute
+def handle_digests(
+    thread_id,
+    discussion,
+    course_id,
+    body,
+    title,
+    url,
+    author_id,
+    author_username,
+    author_email,
+    object_type,
+):
+    """
+    Get the digest subscribers for a thread and notify them.
+
+    Arguments:
+        thread_id (str): The thread id.
+        discussion (dict): The discussion dict.
+        course_id (str): The course id.
+        body (str): The body of the post.
+        title (str): The title of the post.
+        url (str): The url of the post.
+        author_id (str): The author id.
+        author_username (str): The author username.
+        author_email (str): The author email.
+        object_type (str): The forum object type.
+    """
+    digest_preferences = ForumNotificationPreference.objects.filter(
+        preference__in=(
+            PreferenceOptions.ALL_POSTS_DAILY_DIGEST,
+            PreferenceOptions.ALL_POSTS_WEEKLY_DIGEST,
+        ),
+        course_id=course_id,
+    ).select_related("user")
+
+    for preference in digest_preferences:
+        digest, _ = ForumNotificationDigest.objects.get_or_create(
+            user=preference.user,
+            course_id=course_id,
+            defaults={
+                "threads_json": "[]",
+                "digest_type": preference.preference,
+            },
         )
+        threads_json = json.loads(digest.threads_json)
+        threads_json.append(
+            {
+                "thread_id": thread_id,
+                "discussion": discussion,
+                "body": body,
+                "title": title,
+                "url": url,
+                "author_id": author_id,
+                "author_username": author_username,
+                "author_email": author_email,
+                "object_type": object_type,
+            }
+        )
```

Comment:
should we do something here with old digests? As I mentioned [here](https://github.com/eduNEXT/platform-plugin-forum-email-notifier/pull/4/files#r1346425137). Or that's not a worry?

Reply:
`handle_digest` just stores the information of the updates for the subscribed users and its content is deleted every time the email is digested

Reply:
https://github.com/eduNEXT/platform-plugin-forum-email-notifier/pull/4/files#r1347361202